### ISSUE TITLE
Content checksum did not match depending on data

### ISSUE THREAD

**Issue by IvanAntipov** [IonKiwi/lz4.net#4]
Check sum did not match on some data.

```
//Write Data
var tmpFileName = baddata.txt; // attached to issue
var bad = File.ReadAllBytes(tmpFileName)
var f = File.Create(@"C://tmp/baddata3.lz4")
var comp = LZ4Stream.CreateCompressor(f,LZ4StreamMode.Write);
comp.Write(bad,0,bad.Length);
comp.Close();
f.Close();
```

```
//Read Data
using (var stream = LZ4Stream.CreateDecompressor(File.OpenRead(@"C://tmp/baddata3.lz4"), LZ4StreamMode.Read))
{
        var dummy = new byte[16000000];
        while (stream.Read(dummy,0,dummy.Length) >0)
         {
         }
}
```

```
System.Exception: Content checksum did not match
   at lz4.LZ4Stream.AcquireNextBlock()
   at lz4.LZ4Stream.Read(Byte[] buffer, Int32 offset, Int32 count)
```

[baddata.txt](https://github.com/IonKiwi/lz4.net/files/810557/baddata.txt)

lz4.net Version 1.0.7 and 1.0.6

Lz4 windows application (LZ4 v1.7.5) decompress baddata3.lz4 normaly (command lz4.exe -t pass successfully )

**Comment by IvanAntipov** [comment]
The code below does not reproduce the problem. (The problem is still actual on production)
I will reopen the issue if I manage to create reproducible example

**Comment by IvanAntipov** [comment]
Example fixed to be reproducible

**Comment by IonKiwi** [comment]
Thanks for reporting this and the reproducible example!

Found an error in the code, when using linked block mode (default mode).
There are some data blocks in your example that are incompressible, those blocks are included as-is.
The code failed to reset the stream or save the dictionary in that case.

**Comment by IonKiwi** [comment]
should be fixed with v1.0.8.1
